// @enocean-js/pretty-printer/src/pretty-printer.js
const RESET = '\x1b[0m'
// const BRIGHT = '\x1b[1m'
const DIM = '\x1b[2m'
function toHex (val, len = 2) {
  return val.toString(16).padStart(len, '0')
}
function col (num) {
  return `\x1b[38;5;${num}m`
}
function colored (value, color) {
  return `${col(color)}${value}${RESET}`
}
function dim (value) {
  return `${DIM}${value}${RESET}`
}

const DATA_COLOR = 39
const OPTINAL_DATA_COLOR = 83

const packetTypeColors = {
  0x01: 99,
  0x02: 105,
  0x05: 208
}

// const returnCodeColors = {
//   0x00: 83,
//   0x01: 203,
//   0x02: 203,
//   0x03: 203,
//   0x04: 203,
//   0x05: 203,
//   0x06: 203,
//   0x07: 203,
//   0x82: 203,
//   0x90: 203,
//   0x91: 203
// }

function colorData (packet) {
  return `${colored(toHex(packet.data), DATA_COLOR)}`
}
export const pretty = Object.create({})
pretty.logESP3 = function (packet) {
  if (typeof window === 'undefined') {
    console.log(`${dim(55)}.${colored(toHex(packet.dataLength, 4), DATA_COLOR)}.${colored(toHex(packet.optionalLength), OPTINAL_DATA_COLOR)}.${colored(toHex(packet.packetType), packetTypeColors[packet.packetType])}.${dim(toHex(packet.crc8Header))}.${colorData(packet)}.${colored(packet.optionalLength === 0 ? '' : toHex(packet.optionalData), OPTINAL_DATA_COLOR)}.${dim(toHex(packet.crc8Data))}`)
  } else {

  }
}
