// @enocean-js/eep-transcoder/src/eep/a5-13-01.js
export const a51301 = {
  number: '0x01',
  title: 'Weather Station',
  status: 'released',
  description: 'A receiver that accepts EEP A5-13-01 at teach-in automatically needs to accept telegrams from\n            the same ID that comply to the definitions of EEP A5-13-02 thru EEP A5-13-06. Different\n            telegrams received from that ID need to be distinguished by their 4 bit identifiers.',
  case: [{
    datafield: [{
      data: 'LRN Bit',
      shortcut: 'LRNB',
      description: 'LRN Bit',
      bitoffs: '28',
      bitsize: '1',
      enum: {
        item: [{
          value: '0',
          description: 'Teach-in telegram'
        }, {
          value: '1',
          description: 'Data telegram'
        }]
      }
    }, {
      data: 'Dawn sensor',
      shortcut: 'DWS',
      description: 'Dawn sensor',
      info: 'DB_3 Dawn sensor 0 … 999lx, linear n=0…255',
      bitoffs: '0',
      bitsize: '8',
      range: {
        min: '0',
        max: '255'
      },
      scale: {
        min: '0',
        max: '999'
      },
      unit: 'lx'
    }, {
      data: 'Temperature',
      shortcut: 'TMP',
      description: 'Outdoor Temp',
      info: 'DB_2: Outdoor Temp. -40°C ... +80°C, linear n=0…255',
      bitoffs: '8',
      bitsize: '8',
      range: {
        min: '0',
        max: '255'
      },
      scale: {
        min: '-40',
        max: '+80'
      },
      unit: '°C'
    }, {
      data: 'Wind speed',
      shortcut: 'WND',
      description: 'Wind speed',
      info: 'DB_1: Wind speed 0 ... 70m/s, linear n=0…255',
      bitoffs: '16',
      bitsize: '8',
      range: {
        min: '0',
        max: '255'
      },
      scale: {
        min: '0',
        max: '70'
      },
      unit: 'm/s'
    }, {
      data: 'Identifier',
      shortcut: 'ID',
      description: 'Identifier',
      info: 'DB_0.BIT_7 … 4: Identifier',
      bitoffs: '24',
      bitsize: '4',
      enum: {
        item: {
          value: '0x1',
          description: {}
        }
      }
    }, {
      data: 'Day / Night',
      shortcut: 'D/N',
      description: 'Day / Night',
      info: 'DB_0.BIT_2: Day / Night',
      bitoffs: '29',
      bitsize: '1',
      enum: {
        item: [{
          value: '0',
          description: 'Day'
        }, {
          value: '1',
          description: 'Night'
        }]
      }
    }, {
      data: 'Rain Indication',
      shortcut: 'RAN',
      description: 'Rain Indication',
      info: 'DB_0.BIT_1: Rain Indication',
      bitoffs: '30',
      bitsize: '1',
      enum: {
        item: [{
          value: '0',
          description: 'No Rain'
        }, {
          value: '1',
          description: 'Rain'
        }]
      }
    }, {
      reserved: {},
      data: {},
      shortcut: {},
      description: {},
      info: {},
      bitoffs: '31',
      bitsize: '1'
    }]
  }],
  originalIndex: 107,
  eep: 'a5-13-01',
  rorg_title: '4BS Telegram',
  rorg_number: '0xA5',
  func_title: 'Environmental Applications',
  func_number: '0x13',
  submitter: []
}
