// @enocean-js/eep-transcoder/src/eep/a5-13-03.js
export const a51303 = {
  number: '0x03',
  title: 'Date Exchange',
  status: 'released',
  description: '',
  case: [{
    datafield: [{
      reserved: {},
      bitoffs: '0',
      bitsize: '3'
    }, {
      reserved: {},
      bitoffs: '8',
      bitsize: '4'
    }, {
      reserved: {},
      bitoffs: '16',
      bitsize: '1'
    }, {
      reserved: {},
      bitoffs: '29',
      bitsize: '2'
    }, {
      data: 'LRN Bit',
      shortcut: 'LRNB',
      description: 'LRN Bit',
      bitoffs: '28',
      bitsize: '1',
      enum: {
        item: [{
          value: '0',
          description: 'Teach-in telegram'
        }, {
          value: '1',
          description: 'Data telegram'
        }]
      }
    }, {
      data: 'Day',
      shortcut: 'DY',
      description: 'Day',
      info: 'DB_3.BIT_4 … 0: Day n=1 … 31',
      bitoffs: '3',
      bitsize: '5',
      range: {
        min: '1',
        max: '31'
      },
      scale: {
        min: '1',
        max: '31'
      },
      unit: 'N/A'
    }, {
      data: 'Month',
      shortcut: 'MTH',
      description: 'Month (1->January)',
      info: 'DB_2.BIT_3 … 0: Month n=1 … 12 1->January',
      bitoffs: '12',
      bitsize: '4',
      range: {
        min: '1',
        max: '12'
      },
      scale: {
        min: '1',
        max: '12'
      },
      unit: 'N/A'
    }, {
      data: 'Year',
      shortcut: 'YR',
      description: 'Year (0->Year 2000)',
      info: 'DB_1.BIT_6 … 0: Year n=0 … 99',
      bitoffs: '17',
      bitsize: '7',
      range: {
        min: '0',
        max: '99'
      },
      scale: {
        min: '2000',
        max: '2099'
      },
      unit: 'N/A'
    }, {
      data: 'Identifier',
      shortcut: 'ID',
      description: 'Identifier',
      info: 'DB_0.BIT_7 … 4: Identifier',
      bitoffs: '24',
      bitsize: '4',
      enum: {
        item: {
          value: '0x3',
          description: {}
        }
      }
    }, {
      data: 'Source',
      shortcut: 'SRC',
      description: 'Source',
      info: 'DB_0.BIT_0: Source',
      bitoffs: '31',
      bitsize: '1',
      enum: {
        item: [{
          value: '0',
          description: 'Real Time Clock'
        }, {
          value: '1',
          description: 'GPS or equivalent (e.g. DCF77, WWV)'
        }]
      }
    }]
  }],
  originalIndex: 109,
  eep: 'a5-13-03',
  rorg_title: '4BS Telegram',
  rorg_number: '0xA5',
  func_title: 'Environmental Applications',
  func_number: '0x13',
  submitter: [
    'Elsner electronics'
  ]
}
