// @enocean-js/eep-transcoder/src/eep/a5-13-02.js
export const a51302 = {
  number: '0x02',
  title: 'Sun Intensity',
  status: 'released',
  description: '',
  case: [{
    datafield: [{
      data: 'LRN Bit',
      shortcut: 'LRNB',
      description: 'LRN Bit',
      bitoffs: '28',
      bitsize: '1',
      enum: {
        item: [{
          value: '0',
          description: 'Teach-in telegram'
        }, {
          value: '1',
          description: 'Data telegram'
        }]
      }
    }, {
      data: 'Sun – West',
      shortcut: 'SNW',
      description: 'Sun - West,linear',
      info: 'DB_3: Sun – West 1klx … 150klx, linear n=0…255',
      bitoffs: '0',
      bitsize: '8',
      range: {
        min: '0',
        max: '255'
      },
      scale: {
        min: '0',
        max: '150'
      },
      unit: 'klx'
    }, {
      data: 'Sun – South',
      shortcut: 'SNS',
      description: 'Sun - South,linear',
      info: 'DB_2: Sun – West 1klx … 150klx, linear n=0…255',
      bitoffs: '8',
      bitsize: '8',
      range: {
        min: '0',
        max: '255'
      },
      scale: {
        min: '0',
        max: '150'
      },
      unit: 'klx'
    }, {
      data: 'Sun – East',
      shortcut: 'SNE',
      description: 'Sun - East,linear',
      info: 'DB_1: Sun – East 1klx … 150klx, linear n=0…255',
      bitoffs: '16',
      bitsize: '8',
      range: {
        min: '0',
        max: '255'
      },
      scale: {
        min: '0',
        max: '150'
      },
      unit: 'klx'
    }, {
      data: 'Identifier',
      shortcut: 'ID',
      description: 'Identifier',
      info: 'DB_0.BIT_7 … 4: Identifier',
      bitoffs: '24',
      bitsize: '4',
      enum: {
        item: {
          value: '0x2',
          description: {}
        }
      }
    }, {
      data: 'Hemisphere',
      shortcut: 'HEM',
      description: '0 = north / 1 = south, then swith Sun south to Sun North when in southern hemisphere',
      info: {},
      bitoffs: '29',
      bitsize: '1',
      enum: {
        item: [{
          value: '0',
          description: 'North'
        }, {
          value: '1',
          description: 'South'
        }]
      }
    }, {
      reserved: {},
      data: {},
      shortcut: {},
      description: 'Not Used',
      info: {},
      bitoffs: '30',
      bitsize: '2'
    }]
  }],
  originalIndex: 108,
  eep: 'a5-13-02',
  rorg_title: '4BS Telegram',
  rorg_number: '0xA5',
  func_title: 'Environmental Applications',
  func_number: '0x13',
  submitter: [
    'Elsner electronics'
  ]
}
