// @enocean-js/eep-transcoder/src/eep/a5-13-06.js
export const a51306 = {
  number: '0x06',
  title: 'Geographic Position Exchange',
  status: 'released',
  case: [{
    datafield: [{
      reserved: {},
      bitoffs: '29',
      bitsize: '3'
    }, {
      data: 'LRN Bit',
      shortcut: 'LRNB',
      description: 'LRN Bit',
      bitoffs: '28',
      bitsize: '1',
      enum: {
        item: [{
          value: '0',
          description: 'Teach-in telegram'
        }, {
          value: '1',
          description: 'Data telegram'
        }]
      }
    }, {
      data: 'Latitude',
      shortcut: 'LAT',
      description: 'Latitude',
      info: 'combined Latitude from MSB+LSB',
      spread: [
        {
          bitoffs: 0,
          bitsize: 4
        }, {
          bitoffs: 8,
          bitsize: 8
        }
      ],
      range: {
        min: 0,
        max: 4095
      },
      scale: {
        min: -90,
        max: 90
      },
      unit: '°'
    }, {
      data: 'Longitude',
      shortcut: 'LOT',
      description: 'Longitude',
      info: 'combined Longitude from MSB+LSB',
      spread: [
        {
          bitoffs: 4,
          bitsize: 4
        },
        {
          bitoffs: 16,
          bitsize: 8
        }
      ],
      range: {
        min: 0,
        max: 4095
      },
      scale: {
        min: -180,
        max: 180
      },
      unit: '°'
    }, {
      data: 'Identifier',
      shortcut: 'ID',
      description: 'Identifier',
      info: 'DB_0.BIT_7 … 4: Identifier',
      bitoffs: '24',
      bitsize: '4',
      enum: {
        item: {
          value: '0x6',
          description: {}
        }
      }
    }]
  }],
  originalIndex: 112,
  eep: 'a5-13-06',
  rorg_title: '4BS Telegram',
  rorg_number: '0xA5',
  func_title: 'Environmental Applications',
  func_number: '0x13',
  description: '',
  submitter: []
}
