// @enocean-js/eep-transcoder/src/eep/a5-13-04.js
export const a51304 = {
  number: '0x04',
  title: 'Time and Day Exchange',
  status: 'released',
  description: '\n          <br/>\n          <br/>\n          Recommendation: always transmit time in 24 hrs format',
  case: [{
    datafield: [{
      reserved: {},
      bitoffs: '8',
      bitsize: '2'
    }, {
      reserved: {},
      bitoffs: '16',
      bitsize: '2'
    }, {
      data: 'LRN Bit',
      shortcut: 'LRNB',
      description: 'LRN Bit',
      bitoffs: '28',
      bitsize: '1',
      enum: {
        item: [{
          value: '0',
          description: 'Teach-in telegram'
        }, {
          value: '1',
          description: 'Data telegram'
        }]
      }
    }, {
      data: 'Weekday',
      shortcut: 'WDY',
      description: 'Weekday (1 -> Monday)',
      info: 'DB_3.BIT_7 … 5: Weekday n=1 … 7 1 -> Monday',
      bitoffs: '0',
      bitsize: '3',
      enum: {
        item: [{
          value: '1',
          description: 'Monday'
        }, {
          value: '2',
          description: 'Tuesday'
        }, {
          value: '3',
          description: 'Wednesday'
        }, {
          value: '4',
          description: 'Thursday'
        }, {
          value: '5',
          description: 'Friday'
        }, {
          value: '6',
          description: 'Saturday'
        }, {
          value: '7',
          description: 'Sunday'
        }]
      }
    }, {
      data: 'Hour',
      shortcut: 'HR',
      description: 'Hour',
      info: 'DB_3.BIT_4 … 0: Hour n=0 … 23',
      bitoffs: '3',
      bitsize: '5',
      range: {
        min: '0',
        max: '23'
      },
      scale: {
        min: '0',
        max: '23'
      },
      unit: 'N/A'
    }, {
      data: 'Minute',
      shortcut: 'MIN',
      description: 'Minute',
      info: 'DB_2.BIT_5 … 0: Minute n=0 … 59',
      bitoffs: '10',
      bitsize: '6',
      range: {
        min: '0',
        max: '59'
      },
      scale: {
        min: '0',
        max: '59'
      },
      unit: 'N/A'
    }, {
      data: 'Second',
      shortcut: 'SEC',
      description: 'Second',
      info: 'DB_1.BIT_5 … 0: Year n=0 … 59',
      bitoffs: '18',
      bitsize: '6',
      range: {
        min: '0',
        max: '59'
      },
      scale: {
        min: '0',
        max: '59'
      },
      unit: 'N/A'
    }, {
      data: 'Identifier',
      shortcut: 'ID',
      description: 'Identifier',
      info: 'DB_0.BIT_7 … 4: Identifier',
      bitoffs: '24',
      bitsize: '4',
      enum: {
        item: {
          value: '0x4',
          description: {}
        }
      }
    }, {
      data: 'Time Format',
      shortcut: 'TMF',
      description: 'Time Format',
      info: 'DB_0.BIT_2: Time Format',
      bitoffs: '29',
      bitsize: '1',
      enum: {
        item: [{
          value: '0',
          description: '24 hours'
        }, {
          value: '1',
          description: '12 hours'
        }]
      }
    }, {
      data: 'AM/PM',
      shortcut: 'A/PM',
      description: 'AM or PM',
      info: 'DB_0.BIT_1: AM/PM',
      bitoffs: '30',
      bitsize: '1',
      enum: {
        item: [{
          value: '0',
          description: 'AM'
        }, {
          value: '1',
          description: 'PM'
        }]
      }
    }, {
      data: 'Source',
      shortcut: 'SRC',
      description: 'Source',
      info: 'DB_0.BIT_0: Source',
      bitoffs: '31',
      bitsize: '1',
      enum: {
        item: [{
          value: '0',
          description: 'Real Time Clock'
        }, {
          value: '1',
          description: 'GPS or equivalent (e.g. DCF77, WWV)'
        }]
      }
    }]
  }],
  originalIndex: 110,
  eep: 'a5-13-04',
  rorg_title: '4BS Telegram',
  rorg_number: '0xA5',
  func_title: 'Environmental Applications',
  func_number: '0x13',
  submitter: [
    'Elsner electronics'
  ]
}
