// @enocean-js/common-command/src/common-commands/CO_RD_VERSION.js
export const CO_RD_VERSION = {
  name: 'CO_RD_VERSION',
  commandCode: 3,
  responsDefinition: {
    0: [
      {
        name: 'appVersion',
        location: 'data',
        offset: 1,
        length: 4,
        retFunc: x => {
          var s = x.toString('hex')
          return `${parseInt(s.substr(0, 2), 'hex')}.${parseInt(s.substr(2, 2), 'hex')}.${parseInt(s.substr(4, 2), 'hex')}.${parseInt(s.substr(6, 2), 'hex')}`
        }
      }, {
        name: 'apiVersion',
        location: 'data',
        offset: 5,
        length: 4,
        retFunc: x => {
          var s = x.toString('hex')
          return `${parseInt(s.substr(0, 2), 'hex')}.${parseInt(s.substr(2, 2), 'hex')}.${parseInt(s.substr(4, 2), 'hex')}.${parseInt(s.substr(6, 2), 'hex')}`
        }
      }, {
        name: 'chipId',
        location: 'data',
        offset: 9,
        length: 4,
        retFunc: x => `0x${x.toString('hex')}`
      }, {
        name: 'chipVersion',
        location: 'data',
        offset: 13,
        length: 4,
        retFunc: x => `0x${x.toString('hex')}`
      }, {
        name: 'appDescription',
        location: 'data',
        offset: 17,
        length: 16,
        retFunc: x => x.filter(item => item > 0).map(item => String.fromCharCode(item)).join('')
      }
    ]
  }
}
