// @enocean-js/esp3-packets/src/ESP3Packet.js
import { toCRC8 } from '@enocean-js/crc8'
import ByteArray from '@enocean-js/byte-array'

export const makeESP3Packet = function (input) {
  let packetBytes = ByteArray.from(input)
  function setDataLength (value) {
    var byte0 = value % 256
    var byte1 = (value - byte0) / 256
    packetBytes.set([byte1, byte0], 1)
  }
  return Object.freeze({
    push (...value) {
      packetBytes.push(...value)
    },
    shift () {
      packetBytes.shift()
    },
    slice (...args) {
      packetBytes.slice(...args)
    },
    findIndex (func) {
      return packetBytes.findIndex(func)
    },
    getItem (index) {
      return packetBytes[index]
    },
    setItem (index, value) {
      packetBytes[index] = value
    },
    clear () {
      packetBytes = new ByteArray()
    },
    get length () {
      return packetBytes.length
    },
    get dataLength () {
      return packetBytes[1] * 256 + packetBytes[2]
    },
    get optionalLength () {
      return packetBytes[3]
    },
    get packetType () {
      return packetBytes[4]
    },
    get header () {
      return packetBytes.slice(1, 5)
    },
    get data () {
      return packetBytes.slice(6, 6 + this.dataLength)
    },
    set data (value) {
      value = ByteArray.from(value)
      var opt = this.optionalData
      packetBytes.splice(6, this.dataLength, ...value)
      packetBytes.splice(6 + value.length, this.optionalLength, ...opt)
      packetBytes.length = 7 + value.length + this.optionalLength
      setDataLength(value.length)
      this.fixPacket()
    },
    get optionalData () {
      return packetBytes.slice(6 + this.dataLength, 6 + this.dataLength + this.optionalLength)
    },
    get body () {
      return packetBytes.slice(6, 6 + this.dataLength + this.optionalLength)
    },
    isHeaderOK () {
      return this.header.reduce(toCRC8, 0) === packetBytes[5]
    },
    isBodyOK () {
      return this.body.reduce(toCRC8, 0) === packetBytes[6 + this.dataLength + this.optionalLength]
    },
    isPacketOK () {
      return this.isHeaderOK() && this.isBodyOK()
    },
    fixPacket () {
      packetBytes[5] = this.header.reduce(toCRC8, 0)
      packetBytes[6 + this.dataLength + this.optionalLength] = this.body.reduce(toCRC8, 0)
    },
    toString (encoding) {
      return packetBytes.toString(encoding)
    },
    toJSON () {
      return {
        syncByte: 0x55,
        header: {
          dataLength: this.dataLength,
          optionalLength: this.optionalLength,
          packetType: this.packetType
        },
        crc8Header: packetBytes[5],
        data: this.data,
        optionalData: this.optionalData,
        crc8Data: packetBytes[6 + this.dataLength + this.optionalLength]
      }
    }
  })
}
export default makeESP3Packet
